### ISSUE TITLE
[Bug][SegFault] SQLite Database instance must be referenced in class attribute even if only Statements are used

### ISSUE THREAD

**Issue by lvlingxiao1** [oven-sh/bun#13143]
### How can we reproduce the crash?

This is somewhat hard to reproduce. I have a class similar to

```
export class MyRepositorySQLite {
	#putItemStatement : Statement<Entry, [string]>;
	constructor(DB_PATH: string) {
		const db = new Database(DB_PATH, { create: true });
		this.#putItemStatement = db.prepare('update items set value = ? where id = ?');
...
	putItem(id: string, value: string) {
		this.#putItemStatement.run(value, id);
	}
...
```

When I test this alone it is running fine, however, when I integrate with my whole application, it started to crash due to Segmentation fault.

I debugged for a while, and found that I change the class to

```
export class MyRepositorySQLite {
	#db: Database;
	#putItemStatement : Statement<Entry, [string]>;
	constructor(DB_PATH: string) {
		this.db = new Database(DB_PATH, { create: true });
		this.#putItemStatement = this.db.prepare('update notes set value = ? where id = ?');
...
```

Notice that in the original code `db` is only a local variable and in the code after I added it to a class attribute. This fixed the issue somehow... what I guess is the garbage collection cleaned up the Database instance at some point (e.g. when I have used enough memory) then the Statement will start to crash, I don't know for sure.

The same issue happened on my Windows desktop with WSL2 and my Raspberry PI with Ubuntu 22.04.4.

Could you please take a look at this issue?

### Relevant log output

```shell
2024-08-07T02:12:49: ============================================================
2024-08-07T02:12:49: Bun v1.1.21 (70ca2b76) Linux arm64
2024-08-07T02:12:49: 5.15.0 | glibc v2.35
2024-08-07T02:12:49: Args: "/home/lvlingxiao/.bun/bin/bun" "--harmony" "/var/www/node/apps/app.ts"
2024-08-07T02:12:49: Features: jsc Bun.stdin(2) http_server transpiler_cache(2) tsconfig(12)
2024-08-07T02:12:49: Builtins: "bun:main" "bun:sqlite" "detect-libc" "node:assert" "node:async_hooks" "node:buffer" "node:child_process" "node:constants" "node:crypto" "node:dns" "node:events" "node:fs" "node:http" "node:https" "node:net" "node:os" "node:path" "node:process" "node:querystring" "node:stream" "node:string_decoder" "node:tls" "node:tty" "node:url" "node:util" "node:util/types" "node:zlib" "node:punycode" "node:cluster"
2024-08-07T02:12:49: Elapsed: 12467105ms | User: 9038ms | Sys: 3204ms
2024-08-07T02:12:49: RSS: 0.92GB | Peak: 121.48MB | Commit: 0.92GB | Faults: 71
2024-08-07T02:12:49:
2024-08-07T02:12:49: panic(main thread): Segmentation fault at address 0x80
2024-08-07T02:12:49: oh no: Bun has crashed. This indicates a bug in Bun, not your code.
2024-08-07T02:12:49:
2024-08-07T02:12:49: To send a redacted crash report to Bun's team,
2024-08-07T02:12:49: please file a GitHub issue using the link below:
2024-08-07T02:12:49:
2024-08-07T02:12:49:  https://bun.report/1.1.21/La170ca2b7AqgghwCupk++D29Du4xk4D_urx2xDurx2xDm+o2xD_urx2xD_A2AgI
```

### Stack Trace (bun.report)

Bun v1.1.21 ([`70ca2b7`](<https://github.com/oven-sh/bun/tree/70ca2b76c3d33f7ceeea43a702b396ab8ffce25b>)) on linux aarch64 [AutoCommand]

Segmentation fault at address 0x00000080

- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/70ca2b76c3d33f7ceeea43a702b396ab8ffce25b/ld-temp.o#L0>): `WTF::jscSignalHandler`
- [`mod.rs:542`](<https://github.com/oven-sh/bun/blob/70ca2b76c3d33f7ceeea43a702b396ab8ffce25b/src/ptr/mod.rs#L542>): `??`
- [`JSSQLStatement.cpp:2228`](<https://github.com/oven-sh/bun/blob/70ca2b76c3d33f7ceeea43a702b396ab8ffce25b/src/bun.js/bindings/sqlite/JSSQLStatement.cpp#L2228>): `WebCore::jsSQLStatementExecuteStatementFunctionRun`
- *1 unknown/js code*
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/70ca2b76c3d33f7ceeea43a702b396ab8ffce25b/ld-temp.o#L0>): `llint_op_call_ignore_result`
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/70ca2b76c3d33f7ceeea43a702b396ab8ffce25b/ld-temp.o#L0>): `llint_op_call_ignore_result`
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/70ca2b76c3d33f7ceeea43a702b396ab8ffce25b/ld-temp.o#L0>): `llint_op_call`
- *1 unknown/js code*
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/70ca2b76c3d33f7ceeea43a702b396ab8ffce25b/ld-temp.o#L0>): `llint_op_call_ignore_result`
- *1 unknown/js code*

**Comment by Jarred-Sumner** [comment]
Can you run `bun upgrade --canary` and try again? This might've been fixed in https://github.com/oven-sh/bun/pull/13105

**Comment by Jarred-Sumner** [comment]
I believe this was fixed in Bun v1.1.22

If you're still running into it, please leave a comment
